hydromt/components/vector.py

```diff
@@ -65,10 +66,16 @@ def data(self) -> xr.Dataset:
         assert self._data is not None
         return self._data
 
+    @property
+    def _region_data(self) -> Optional[gpd.GeoDataFrame]:
+        raise AttributeError(
+            "region cannot be found in vector component. Meaning that the region_component is not set or could not be found in the model."
+        )
```

Comment:
This should implement: https://github.com/Deltares/hydromt/blob/f1fad0ef457ce9623fb5969f52bd8bf6f28140f6/hydromt/models/model_vector.py#L376

Reply:
Also check test_equal for vector.